src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/location/LocationTest.java

```diff
@@ -0,0 +1,137 @@
+
+        final Response responseUploadLocations = doPostRequestMultiPart(
+            UPLOAD_LOCATIONS_URL,
+            headerMapUploadLocations, LOCATION_LIST, csvFile
+        );
+
+        assertThat(responseUploadLocations.getStatusCode()).isEqualTo(OK.value());
+        assertEquals(responseUploadLocations.as(Location[].class).length, 3,
+                     "Failed to create all locations");
+        assertThat(responseUploadLocations.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseGetLocationById = doGetRequest(
+            GET_LOCATION_BY_ID_URL,
+            headerMap
+        );
+        assertThat(responseGetLocationById.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetLocationById.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseGetAllLocations = doGetRequest(
+            BASE_LOCATIONS_URL,
+            headerMap
+        );
+        assertThat(responseGetAllLocations.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetAllLocations.asString().contains(TEST_LOCATION_NAME));
+        assertThat(responseGetAllLocations.asString().contains("TestLocationTwo"));
+        assertThat(responseGetAllLocations.asString().contains("TestLocationThree"));
+
+        final Response responseGetLocationByName = doGetRequest(
+            GET_LOCATION_BY_NAME_URL,
+            headerMap
+        );
+        assertThat(responseGetLocationByName.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetLocationByName.asString().contains("2001"));
+
+
+        final Response responseGetLocationByRegionAndJurisdiction =
+            given()
+                .relaxedHTTPSValidation()
+                .headers(getRequestHeaders(headerMap))
+                .queryParam("jurisdictions", List.of(TEST_JURISDICTION1, TEST_JURISDICTION2))
+                .queryParam("regions", List.of(TEST_REGION))
+                .when()
+                .get(GET_LOCATION_BY_REGION_AND_JURISDICTION)
+                .thenReturn();
+
+        assertThat(responseGetLocationByRegionAndJurisdiction.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseGetLocationByRegionAndJurisdiction.asString().contains(TEST_LOCATION_NAME));
+
+        final Response responseDownloadCsv = doGetRequest(
+            DOWNLOAD_CSV_LOCATIONS_URL,
+            headerMap
+        );
+        assertThat(responseDownloadCsv.getStatusCode()).isEqualTo(OK.value());
+        assertThat(responseDownloadCsv.asString().contains(TEST_LOCATION_NAME));
+        assertThat(responseDownloadCsv.asString().contains("TestLocationTwo"));
+        assertThat(responseDownloadCsv.asString().contains("TestLocationThree"));
+
+        final Response responseDeleteLocationById = doDeleteRequest(
+            GET_LOCATION_BY_ID_URL,
+            deleteHeaderMap
+        );
+        assertThat(responseDeleteLocationById.getStatusCode()).isEqualTo(OK.value());
```

Comment:
When this PR is merged to master, this test is going to run on staging. Don't we need to clean up the locations on staging after the test is run? Currently 2 locations remain

Reply:
Fixed